src/Core/Grid/Presenter/GridPresenter.php

```diff
@@ -169,6 +171,27 @@ private function getColumnFilters(GridDefinitionInterface $definition)
         return $columnFiltersMapping;
     }
 
+    /**
+     * Get filters that have associated columns.
+     *
+     * @param GridDefinitionInterface $definition
+     *
+     * @return array
+     */
+    private function getHiddenColumnFilters(GridDefinitionInterface $definition)
```

Comment:
```suggestion
    private function getHiddenColumnFilters(GridDefinitionInterface $definition): array
```

Reply:
So originally I did this because of this part.

```
 {% for column in grid.columns %}
    <td
      data-type="{{ column.type }}"
      data-column-id="{{ column.id }}"
    >
      {% if loop.first and grid.actions.bulk|length > 0 %}
        {{ include('@PrestaShop/Admin/Common/Grid/Blocks/bulk_actions_select_all.html.twig', {'grid': grid}) }}
      {% endif %}

      {% if grid.filter_form|length > 1 %}
        {% if grid.column_filters[column.id] is defined %}
          {% for filter_name in grid.column_filters[column.id] %}
            {{ form_widget(grid.filter_form[filter_name]) }}
          {% endfor %}
        {% endif %}
      {% endif %}
    </td>
  {% endfor %}
```

Essentially you don't just go trough all filters and rendered them, but you go trough all columns and render filters for those columns. The column for hidden filter is not rendered so it won't appear in that list. But now I tested and strangely after I removed my rendering for hidden filters the whole thing still worked.

Well turns out I missed one key thing, there is a {{ form_end(grid.filter_form) }} at the end of the table, so it will render all hiddenFilters at the on of the table. 

So to sum up I will remove the whole separate rendering for hidden filters.